js/okx.js

```diff
@@ -2252,7 +2299,12 @@ module.exports = class okx extends Exchange {
         if (limit !== undefined) {
             request['limit'] = limit; // default 100, max 100
         }
-        const response = await this.privateGetTradeOrdersPending (this.extend (request, params));
+        let method = 'privateGetTradeOrdersPending';
+        const ordType = this.safeString (params, 'ordType');
+        if (ordType === 'trigger') {
```

Comment:
I would suggest to map all "algo" orders in 'options' (conditional, oco, trigger, etc):

- https://www.okx.com/docs-v5/en/#rest-api-trade-get-algo-order-list

And then doing a lookup like this:

```JavaScript
const options = this.safeValue (this.options, 'fetchOpenOrders', {});
const algoOrderTypes = this.safeValue (options, 'algoOrderTypes', {});
if (ordType in algoOrderTypes) {
    method = 'privateGetTradeOrdersAlgoPending';
}
```

Reply:
We should set this up like ftx and gateio. If there is a param `stop` within params, then we're looking at stop orders

Reply:
and then just an `orderType` param, and you can check `if (stop || (orderType !== undefined)) { method = 'privateGetTradeOrdersAlgoPending' }`

Let's create a docstring for this method 

```
    /**
     * Returns orders that are still open
     * @param {string} symbol Unified market symbol
     * @param {integer} since Pagination of data to return records newer than
     * @param {integer} limit Number of results per request. The maximum is 100; The default is 100
     * @param {dict} params Extra and exchange specific parameters
     * @param {dict} params.till Pagination of data to return records earlier than
     * @param {dict} params.stop True if fetching trigger orders
     * @param {dict} params.ordType "conditional", "oco", "trigger", "move_order_stop", "iceberg", or "twap"
     * @param {dict} params.algoId Algo ID
     * @returns [An order structure]{@link https://docs.ccxt.com/en/latest/manual.html#order-structure}
     */
```